### ISSUE TITLE
[Tier 3: Medium] Users are not able to update their address on VA.gov Profile

### ISSUE THREAD

**Issue by clantosswett** [department-of-veterans-affairs/va.gov-team#6012]
**Issue Description:** Users are not able to update their mailing address on VA.gov. They go to their VA.gov profile, click edit mailing address, add in the new address, click update and receive the below error message: 

_We couldn’t save your recent mailing address update. Please try again later._

I have included whether the user is trying to change their address to an international or domestic address next to the Issue number. The User descriptions go into more detail for each example. Addresses can be provided if needed for investigation. 

**Priority:** Medium

**Total Number of Issues:** 5

**Issue Number and User Description:**

- IRIS: 200110-000191 (Domestic Address)

I have tried a number of times and even called your office and still no one has changed my mailing address. Your website will not let me do it and I've tried a number of times. I called the VA and told them I have a new address and they were suppose to update it but my mail is still going to my old address. Please change this as this is the message I get from your online website:

Mailing address Edit
We couldn’t save your recent mailing address update. Please try again later.

- IRIS: 200106-000153 (Domestic Address)

This is still an ongoing issue with updating my profile and mailing address. 

I click my profile, click my contact information, click edit mailing address and receive the following error message: We couldn’t save your recent mailing address update. Please try again later. I also attempted to add mailing address and get the following error message: We couldn’t save your recent home 
address update. Please try again later. 

URL: https://www.va.gov/profile/#contact-information 

What type of device are you using (i.e. dell laptop, MacBook, iPhone, etc.)? HP Laptop 

What account do you use to sign into VA.gov (DS Logon, Myhealthevet, ID.me)? DS Logon with CAC

- IRIS: 200127-001067 (Domestic Address)

I log on to the site using an I Mac computer and by using my ID me info. 
Every time I attempt to edit both my mailing and residence addresses it says that the info could not be saved due to a system error on the site’s fault and for me to try again later. This happens and has happened for a long time. 

I am trying to change both of my mailing and residence addresses

- IRIS: 200127-001067 (International Address)

Step by step: 
Login to VA.GOV Sign in using iD.ME Complete my sign in with the 6 digits code sent to my phone Click on my PROFILE Click on EDIT my mailing address Try changing address [PII REMOVED] to Save Error message: We couldn’t save your recent mailing address update. Please try again later. 
https://www.va.gov/profile/ 

Currently using MacBook Air Google Chrome browser but also have tried changing using Android phone Samsung Galaxy Note 8 with same error 
message 
This has been going on for over 3 months. I have tried updating at various times and hours.

**Added issue**

**IRIS:200212-000709**

_Thank you for your response
Unfortunately for the past few days i click the link to update my address
and I keep getting the “try again later” text box. Please assist as I
cannot access much on this site or eBenefits_

**Comment by Samara-Strauss** [comment]
@erikphansen do we have any idea why this might be happening?

I'm wondering if this will be fixed with the address validation updates we're going to be launching in a couple weeks. @clantosswett after launch, we should follow up with these folks to see if they are still having issues. If they are, we might want to hop on a call with a couple people so we can see what's happening and troubleshoot for them.

**Comment by erikphansen** [comment]
Yeah I assume they have "bad" addresses (like mine) that are being silently rejected. So the address validation feature should also fix this.

**Comment by erikphansen** [comment]
+1 to following up and/or getting their addresses so we can see what VA Profile thinks of them

**Comment by clantosswett** [comment]
I figured that might be the issue. Let's leave this issue open until we can follow up. Thanks!

**Comment by Samara-Strauss** [comment]
Sounds good. I will add following up with these veterans as a post-launch item to our checklist.

**Comment by Kimberley2019** [comment]
@samara-strauss @erikphansen I have another address issue and am wondering if you would like for me to add it to this ticket or would you like for me to open another one?

**Comment by Samara-Strauss** [comment]
Please open a different ticket if it's a different issue @Kimberley2019. If it's another person with this same issue, feel free to add details here.

Also, friendly reminder not to include any PII on Github, which includes names and addresses (even if you don't pair those pieces of info together). You may already know that, just feels like this always bears repeating.

**Comment by Kimberley2019** [comment]
Hi @Samara-Strauss it appears to be the same issue.  The see below.  I will also add above.

**IRIS:200212-000709**

_Thank you for your response
Unfortunately for the past few days i click the link to update my address
and I keep getting the “try again later” text box. Please assist as I
cannot access much on this site or eBenefits_

**Comment by mshea0606** [comment]
Hi @clantosswett @Kimberley2019 ,
I'm the new PM on the Authenticated Experience team. I'm working on putting a pre-post launch metrics together for the Address Validation feature. I was looking at this ticket, and wanted to see what time frame those 5 issues mentioned are from so I can use it as a baseline and see how that number changes post Address Validation launch.

**Comment by clantosswett** [comment]
@Samara-Strauss I know that the roll out of the address validation feature has begun. Do you think it is worth reaching out to these users to see if the new feature has fixed the issue. Or should we wait until the it is fully out for all users?

**Comment by Samara-Strauss** [comment]
Let's wait until we've launched to 100%, which should be by Monday. I can let you know when we've done that.

**Comment by Kimberley2019** [comment]
Goof Morning @Samara-Strauss I know that you have been working closely with Chante on this, but I wanted to follow up to see if the feature was launched on Monday.   
If she has already followed up on this, then my apologies for being a bother.

**Comment by erikphansen** [comment]
@Kimberley2019 It did! Moving forward it would be tremendously helpful if we could get addresses that Veterans are still unable to save to their profile. We do not log that for privacy reasons, but if individuals are willing to share addresses they can't save that would be great.

**Comment by Samara-Strauss** [comment]
@Kimberley2019 yes, this feature went out. In terms of next steps:

- It would be great if you could reach back out to veterans who've had issues saving their addresses and see if they can try saving them again. I'm hoping the new update we launched will allow some of them to be able to save/update their addresses.
- If people are still not able to save their addresses, it would be great if we could get a couple of these problem addresses to further explore this issue. As @erikphansen mentioned, **please DO NOT put any addresses in this thread, as that info is PII and this is a public repo**. We'll have to figure out how to get that information to us. We may be able to save an address **without** a name attached to it in the va.gov-team/sensitive repo. If not, we'd have to exchange that information over a phone call/Zoom.

**Comment by joanneesteban** [comment]
@clantosswett or @Kimberley2019 will reach out to this group of users to see if this issue was solved.

**Comment by Samara-Strauss** [comment]
@joanneesteban thank you!

@clantosswett @Kimberley2019 I am seeing a huge reduction in the % of mailing address save errors. It's dropped from ~18 – ~20% before we launched this feature to ~6% since we went to 100% of users. I'm optimistic that most of the people who'd written in with concerns will now be able to save their addresses, but if not, let's regroup.

**Comment by erikphansen** [comment]
And I'm kinda hoping we can find addresses that somehow trigger a failure so we can figure out why we still get 6% failures!

**Comment by Samara-Strauss** [comment]
@clantosswett @Kimberley2019 @joanneesteban any update here? Have you been able to reach out to the customers who reported this issue?

**Comment by Kimberley2019** [comment]
@Samara-Strauss I'm on it! :)

**Comment by Samara-Strauss** [comment]
Thank you, @Kimberley2019!! Keep us posted. Looking forward to hearing what you hear back.

**Comment by Kimberley2019** [comment]
Hi @Samara-Strauss I sent a follow up email yesterday.  Thus far, none of the individuals have emailed me back.  Lets keep our fingers crossed that they will.  I will keep you posted.

**Comment by Samara-Strauss** [comment]
Thanks for the update @Kimberley2019. Definitely keep us posted. If we don't hear back, how should we proceed with this ticket? Close it out and open a new one if the issue crops back up? Or something else?

**Comment by Kimberley2019** [comment]
Hi @Samara-Strauss lets give these individuals until EOD Monday 4/20/2020.  I will send another F/U email tomorrow if I have not heard back from anyone today.  If by Tuesday morning I have not heard from anyone, we will close the ticket noting those actions.
If another ticket needs to be opened later down the line, then that is what we can do; I don't think that we need to leave this open if we don't hear back from anyone.

**Comment by Samara-Strauss** [comment]
Sounds good. We can def leave this open for a few days. No rush. We'll close if we don't hear anything in a week or so.

**Comment by Kimberley2019** [comment]
Sounds good to me.  Still nothing as of yet.

**Comment by Kimberley2019** [comment]
@Samara-Strauss I have one person that has emailed me back.  I am still waiting on the others to reply (by the end of the week). The information that he provided is below.  I have removed all PII, but he did provide his address.  I can give you that information through a Slack or Zoom call if you need it.

I also asked him if he would be willing to do a shadowing call if you all needed to conduct one.

200212-000709:

Thank you very much for the follow up. Unfortunately I am still unable to update my contact info including address. When I click on the edit my contact information, I get the same text box that says ‘Unable to do so at this time. Please try again later.’

 

I have only used one address: _PII Remoed_ I’ve had no problems receiving VA mailings at this address. My problem is I can log into both the eBenefiits and VA.gov web sites but I cannot generate the benefits letter. eBenefits gives me a message that says: “ We are unable to generate your letter due to incomplete information contained in our systems”. The VA.gov site does not download my benefits letter. It tells me to change my address in my profile section. When I click on the Change Address link I keep getting this message: We’re sorry. Something went wrong on our end and we couldn’t save the recent updates you made to your profile. Please try again later. I’ve had benefits letters mailed and faxed to me with no problem but I am still unable to do anything through either the eBenefits or VA.gov websites Happy to discuss by phone with any tech support expert 

Vr

**Comment by Samara-Strauss** [comment]
Thanks for this, @Kimberley2019! I appreciate it!

@erikphansen can you review the comment above and let me know what your thoughts are on how to proceed with this? No need to take immediate steps, just let me know how you *would* move forward. My initial thoughts are that we'd need to get this veteran on a video call to see what  exactly is going on in his account. What are your thoughts?

**Comment by erikphansen** [comment]
@Samara-Strauss yes I think we'd need to do a quick screenshare with them to replicate the issue. But based on the error message they reported seeing, it looks like they get past address validation but the save still fails. There are a number of different VA Profile error codes (https://github.com/department-of-veterans-affairs/vets-website/blob/655fca74b24378320141777a409dc351f72a3bb6/src/platform/user/profile/vet360/util/transactions.js#L19-L37) that would trigger the generic error message they are seeing on their Profile.

**Comment by Kimberley2019** [comment]
@Samara-Strauss @erikphansen 
Let me know what you would like for me to do.  If you want me to pass along his contact information so you can reach out directly.

**Comment by Samara-Strauss** [comment]
@erikphansen noted. I will see what I can do to get something set up with this veteran.

@Kimberley2019 have you been directly in touch with this veteran? If so, can you gauge their interest in speaking with us further so we can troubleshoot this error? This would involve us doing a screen share with them so that Erik could look at exactly what the errors are. 

If the veteran seems interested in speaking with us further, then you can **email me** their name and email address.

**Comment by erikphansen** [comment]
@Samara-Strauss do we have a reference that might explain what they VA Profile error codes mean? https://github.com/department-of-veterans-affairs/vets-website/blob/655fca74b24378320141777a409dc351f72a3bb6/src/platform/user/profile/vet360/util/transactions.js#L19-L37

**Comment by Samara-Strauss** [comment]
@erikphansen it looks like we do not. 

@mshea0606 — since we talked about you owning more of the VA Profile relationship, can you please work on getting some documentation together that outlines what those errors mean that @erikphansen linked to above? I can't find any documentation on this in the old repo, so you'll probably have to go to the VA Profile folks to figure out what they mean. Once we have that info, it would be super helpful if you documented that in the /profile folder on Github. Let me know if you have any questions!

**Comment by Kimberley2019** [comment]
@Samara-Strauss The Veteran _is_ willing to do a screensharing.  I will gather his contact information and **email it** to you.

**Comment by Samara-Strauss** [comment]
@Kimberley2019 I just realized you already mentioned he was willing to do a call — apologies that I forgot! All of that sounds good. I will look forward to getting his info. Thank you!!

**Comment by Kimberley2019** [comment]
@Samara-Strauss No worries and no need to apologize at all.
I'll get that information over to you ASAP.

**Comment by erikphansen** [comment]
FWIW it would be good to have @lihanli and/or @sanlouise on this future call as well. It should be simple, and there's likely not much we can do, but the more heads the better.

**Comment by Samara-Strauss** [comment]
@erikphansen sounds good to me!

**Comment by Kimberley2019** [comment]
@Samara-Strauss Just emailed over the information.

**Comment by Samara-Strauss** [comment]
Thank you! I will email the veteran later today.

**Comment by mshea0606** [comment]
@Samara-Strauss @erikphansen I will work on getting the Error code info and creating  updated documentation in Github. "Create updated documentation for Profile error codes #8372" has been created.

**Comment by Samara-Strauss** [comment]
Last Thursday, I emailed the veteran who said they were interested in speaking with us. However, I haven't heard back from him. I'll reach out again in hopes we can get in touch with him.

**Comment by Samara-Strauss** [comment]
Just heard back from the veteran! Going to try to get a call set up with them for tomorrow or Friday.

**Comment by Samara-Strauss** [comment]
Update: Call scheduled for 2:00 ET tomorrow with the veteran.

Also, we were able to try to save the veteran's address and it's perfectly legitimate. We suspect this issue has to do with something else in his record. Hopefully we'll be able to dig to the bottom of this tomorrow.

**Comment by Samara-Strauss** [comment]
@Kimberley2019 have you heard back from any other veterans? Any chance you could reach out again?

We talked to the veteran who had responded to your initial request today. He had an issue where the contact information wasn't showing up at all in his profile, and we have not seen that before. We were not able to get to the bottom of things in the call due to some sign in issues but will hopefully be able to talk to him again now that those appear to be resolved. If we gain some clarity on his issue, I will let you know. But in the meantime, we would love to talk to other folks if we can to determine if they are having a similar issue or experiencing something different entirely.

**Comment by Kimberley2019** [comment]
@Samara-Strauss I have not heard back from any of the other users.  I will reach out again today and keep you posted.

**Comment by Samara-Strauss** [comment]
Thank you!

**Comment by Kimberley2019** [comment]
Good Morning @Samara-Strauss I sent an email out to everyone today, except for 200212-000709, since you have already made contact with him.

If no one responds by the end of the week, please advise on how you would like to proceed with this ticket.

**Comment by Samara-Strauss** [comment]
Thanks @Kimberley2019. 

@erikphansen without talking to users, do we have any way of meaningfully investigating on our end why these problems are still happening? I'm not sure what we can learn from our error logs that might move this forward.

**Comment by Samara-Strauss** [comment]
Update: We are going to try to talk to the veteran from last week again. We have a call tentatively scheduled for tomorrow afternoon.

**Comment by Samara-Strauss** [comment]
Update: We were able to talk to the veteran today. I've created a ticket in our sensitive (private) repo to track our investigation.

https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/issues/101

@Kimberley2019 if we don't hear back form anyone else within a few days, we can close this issue and pursue the investigation in other tickets.

**Comment by Kimberley2019** [comment]
@Samara-Strauss Thank you for the update.  I will close the ticket if I have not heard back from any of the users by Monday.

**Comment by Samara-Strauss** [comment]
Update: From our initial investigation, this definitely seems to be an issue of identity and not the profile itself. 

The good news: that's a lead and may be something affecting other people. The bad news, getting identity issues figured out is (likely) an annoying process that will involve calling the VA and spending lots of time on the phone.

**Comment by Samara-Strauss** [comment]
Since we have not heard back from other veterans, I am going to close this. We definitely improved things with the launch of the address validation feature. For the remaining veterans who are having trouble, we have reason to believe that some of the people who have reported this issue have multiple identities in some of the VA systems we use to determine identity, and this is causing us to be unable to load their profiles.

@Kimberley2019 if we start hearing a lot about this issue again, please let us know.

**Comment by Kimberley2019** [comment]
Thanks @Samara-Strauss Will do!
